tree-sitter-stack-graphs/src/cli/database.rs

```diff
@@ -0,0 +1,42 @@
+// -*- coding: utf-8 -*-
+// ------------------------------------------------------------------------------------------------
+// Copyright © 2021, stack-graphs authors.
+// Licensed under either of Apache License, Version 2.0, or MIT license, at your option.
+// Please see the LICENSE-APACHE or LICENSE-MIT files in this distribution for license details.
+// ------------------------------------------------------------------------------------------------
+
+use anyhow::anyhow;
+use clap::Args;
+use clap::ValueHint;
+use std::path::Path;
+use std::path::PathBuf;
+
+/// CLI arguments for using a database.
+#[derive(Args)]
+pub struct DatabaseArgs {
+    #[clap(
+        long,
+        short = 'D',
+        value_name = "DATABASE_PATH",
+        value_hint = ValueHint::AnyPath,
+        parse(from_os_str),
+    )]
+    pub database: Option<PathBuf>,
+}
+
+impl DatabaseArgs {
+    pub fn default_for_crate(crate_name: &str) -> anyhow::Result<PathBuf> {
```

Comment:
This method is perhaps a bit confusing here, since it does not actually create or act upon `DatabaseArgs` values. It's an associated method to easily get a default database location for a given crate name. I'll move this to be a module level function instead, so that it's clearer that it does not use `DatabaseArgs` in any way.